ml/management/commands/ml_train.py

```
@@ -0,0 +1,210 @@
+- rating : rating provided by a contributor between 2 videos, in [0,100] or [-1,1]
+- score : score of a video outputted by the algorithm, arround [-10, 10]
+- glob, loc : global, local
+- idx : index
+- l_someting : list of someting
+- arr : numpy array
+- tens : torch tensor
+- dic : dictionnary
+- verb : verbosity level
+- VARIABLE_NAME : global variable
+
+Structure:
+- fetch_data() provides data from the database
+- ml_run() uses this data as input, trains via shape_train_predict()
+     and returns video scores
+- save_data() takes these scores and save them to the database
+- these 3 are called by Django at the end of this file
+
+USAGE:
+- define global variables EPOCHS, CRITERIAS
+- set EXPERIMENT_MODE to False
+- set RESUME to True if you want to use previously trained models 
+-- EPOCHS: number of training epochs
+-- CRITERIAS: list of str (there is one training for each criteria)
+- call ml_run(fetch_data()) if you just want the scores in python
+
+"""
+# global variables
+
+EXPERIMENT_MODE = False  # False to compute all data
+
+FOLDER_PATH = "ml/checkpoints/" 
+FILENAME = "models_weights"
+PATH = FOLDER_PATH + FILENAME
+os.makedirs(FOLDER_PATH, exist_ok=True)
+RESUME = False # wether to resume training or not
+
+EPOCHS = 100
+CRITERIAS = ["reliability", "importance", "engaging", "pedagogy", 
+             "layman_friendly", "diversity_inclusion", "backfire_risk", 
+             "better_habits", "entertaining_relaxing"]
+
+def fetch_data():
+    """ Fetches the data from the Comparisons model
+
+    Returns:
+    - comparison_data: list of 
+        [   contributor_id: int, video_id_1: int, video_id_2: int, 
+            criteria: str, score: float, weight: float  ]
+    """
+    comparison_data = [
+        [ccs.comparison.user_id, ccs.comparison.video_1_id, 
+            ccs.comparison.video_2_id, ccs.criteria, ccs.score, ccs.weight]
+        for ccs 
+        in ComparisonCriteriaScore.objects.all().prefetch_related("comparison")]
```

protect this interpreted list in case ccs does not hold a `comparison` element or if (due to a mishap) any other element is missing

---

@lfaucon

---

These fields are supposed to be required. I would prefer not adding the fix that you propose, because if a mishap happens that would allow us to quickly discover it and patch it instead of failing silently